python/ray/serve/_private/http_proxy.py

```diff
@@ -1410,10 +1473,12 @@ async def ready(self):
                     get_component_logger_file_path(),
                 ]
             )
+        elif not self.http_setup_complete.is_set():
```

Comment:
Yea, it is throwing one of them here. There can be 4 outcomes now. Both HTTP and gRPC succeeded, only HTTP succeeded, only gRPC succeeded, and neither HTTP nor gRPC completed. `if self.http_setup_complete.is_set() and self.grpc_setup_complete.is_set()` is when both succeeded. `elif not self.http_setup_complete.is_set()` is when HTTP failed regardless gRPC succeeded or not, it will raise error for that. And the final else case is when only gRPC failed

Reply:
Understand that. What i'm saying is that we should at least log both exceptions before re-throwing

Reply:
I see, let me look into this